presto-jdbc/src/main/java/com/facebook/presto/jdbc/ColumnInfo.java

```diff
@@ -227,6 +233,8 @@ private static int getType(TypeSignature type)
                 return Types.DATE;
             case "decimal":
                 return Types.DECIMAL;
+            case "float":
+                return Types.REAL;
```

Comment:
JDBC's FLOAT = DOUBLE (64b)
JDBC's REAL = FLOAT (32b)

Details: http://docs.oracle.com/javase/1.5.0/docs/guide/jdbc/getstart/mapping.html#1055162
and
https://docs.oracle.com/cd/B19306_01/java.102/b14188/datamap.htm